x-pack/plugins/actions/server/builtin_action_types/cases_webhook/utils.ts

```diff
@@ -0,0 +1,136 @@
+};
+const findTheValue = (obj: Record<string, Record<string, unknown> | unknown>, keys: string[]) => {
+  let currentLevel: unknown = obj;
+  keys.forEach((k: string) => {
+    // @ts-ignore
+    currentLevel = currentLevel[k];
+  });
+  return currentLevel;
+};
+
+export const getObjectValueByKey = (
+  obj: Record<string, Record<string, unknown> | unknown>,
+  key: string
+): string => {
+  try {
+    return findTheValue(obj, splitKeys(key)) as string;
+  } catch (e) {
+    throw new Error(`Value not found in object for key ${key}`);
+  }
+};
+
+export const throwIfResponseIsNotValidSpecial = ({
+  res,
+  requiredAttributesToBeInTheResponse = [],
+}: {
+  res: AxiosResponse;
+  requiredAttributesToBeInTheResponse?: string[];
+}) => {
+  const requiredContentType = 'application/json';
+  const contentType = res.headers['content-type'];
+  const data = res.data;
+
+  /**
+   * Check that the content-type of the response is application/json.
+   * Then includes is added because the header can be application/json;charset=UTF-8.
+   */
+  if (contentType == null) {
+    throw new Error(
+      `Missing content type header in ${res.config.method} ${res.config.url}. Supported content types: ${requiredContentType}`
+    );
+  }
+  if (!contentType.includes(requiredContentType)) {
+    throw new Error(
+      `Unsupported content type: ${contentType} in ${res.config.method} ${res.config.url}. Supported content types: ${requiredContentType}`
+    );
+  }
+
+  /**
+   * Check if the response is a JS object (data != null && typeof data === 'object')
+   * in case the content type is application/json but for some reason the response is not.
+   * Empty responses (204 No content) are ignored because the typeof data will be string and
+   * isObjectLike will fail.
+   * Axios converts automatically JSON to JS objects.
+   */
+  if (isEmpty(data) || !isObjectLike(data)) {
+    throw new Error('Response is not a valid JSON');
+  }
+
+  if (requiredAttributesToBeInTheResponse.length > 0) {
+    const requiredAttributesError = (attrs: string[]) =>
+      new Error(
+        `Response is missing the expected ${attrs.length > 1 ? `fields` : `field`}: ${attrs.join(
+          ', '
+        )}`
+      );
+
+    const errorAttributes: string[] = [];
+    /**
+     * If the response is an array and requiredAttributesToBeInTheResponse
+     * are not empty then we throw an error if we are missing data for the given attributes
+     */
+    requiredAttributesToBeInTheResponse.forEach((attr) => {
+      // Check only for undefined as null is a valid value
+      try {
+        getObjectValueByKey(data, attr);
+      } catch (e) {
+        errorAttributes.push(attr);
+      }
+    });
+    if (errorAttributes.length) {
+      throw requiredAttributesError(errorAttributes);
+    }
+  }
+};
+
+export const removeSlash = (url: string) => (url.endsWith('/') ? url.slice(0, -1) : url);
+
+export const makeCaseStringy = (properties: Record<string, string | string[]>) => ({
```

Comment:
Could we rename this to something like `stringifyObjValues` or `formatForMustache`?